### PR TITLE
net: lib: aws_iot: Add last will topic and message.

### PR DESCRIPTION
**Pull request by simensrostad**
This patch adds the ability to statically configure a last will -topic
and -message that is included in the initial MQTT CONNECT message.
The last will message is published to the last will topic upon an
disconnect from the AWS IoT broker.

### CODE REVIEW

File: subsys/net/lib/aws_iot/Kconfig [hunk 1]
```diff
@@ -78,6 +78,26 @@ config AWS_IOT_TOPIC_DELETE_ACCEPTED_SUBSCRIBE
 config AWS_IOT_TOPIC_DELETE_REJECTED_SUBSCRIBE
 	bool "Subscribe to delete rejected shadow topic, $aws/things/<thing-name>/shadow/delete/rejected"
 
+config AWS_IOT_LAST_WILL
+	bool "This option includes the last will -message and -topic in the initial MQTT handshake"
```
Comment (rlubos) [inline subsys/net/lib/aws_iot/Kconfig:82]: Why `-message` and `-topic` and not simply `message`, `topic`? Same below.

Comment (simensrostad) [reply subsys/net/lib/aws_iot/Kconfig:82]: Well. I guess its not necessary to include the dashes. It was just to emphasize that both "message" and "topic" adheres to "will". But of course, if it looks cleaner omitting the dashes, I can remove them.

Comment (simensrostad) [reply subsys/net/lib/aws_iot/Kconfig:82]: Had some offline discussion about this. Ill change it to not include the dashes. It looks cleaner.

**Review by rlubos — APPROVED** [review]

File: subsys/net/lib/aws_iot/Kconfig [hunk 2]
```diff
@@ -78,6 +78,26 @@ config AWS_IOT_TOPIC_DELETE_ACCEPTED_SUBSCRIBE
...
+
+if AWS_IOT_LAST_WILL
+
+config AWS_IOT_LAST_WILL_MESSAGE
+	string "Message published to the last will topic"
+	default "{\"lastwill\": \"offline\"}"
+
+config AWS_IOT_LAST_WILL_TOPIC
+	string "Topic that the last will message is published to"
+	default "lastwilltopic/"
```
[the remark this answers is not in the document]
Comment (simensrostad) [reply subsys/net/lib/aws_iot/Kconfig]: No. Removed the slash.

**Review by jhn-nordic — APPROVED** [review]

**Review by jtguggedal — APPROVED** [review]

**Pull request merged**